### ISSUE TITLE
Check of all components of Dell PowerEdge R730xd with SSDs gives warning

### ISSUE THREAD

**Issue by quicknick** [dangmocrang/check_idrac#18]
Hi Dung,

thanks for your script, it's so helpfull for me. :-)
when i'm checking all components of a Poweredge R730xd Hardware i receive always an warning state. I think that's because some SSDs where in "PowerStat; 5(!)" (and not e.g. "PowerStat: SPUNUP", like HDDs). You can see my Output at PDisk 16 + 17 (SSDs) by checking all states:

>PS
>--PS 1: OK, Volt I/O: 264 V/(N/A) V, Current: 0.8 A, Watt I/O: 900.0 W/750 W
>--PS 2: OK, Volt I/O: 264 V/(N/A) V, Current: 0.2 A, Watt I/O: 900.0 W/750 W
>FAN
>--System Board Fan1: 3600 RPM - ENABLED/OK
>--System Board Fan2: 3720 RPM - ENABLED/OK
>--System Board Fan3: 3600 RPM - ENABLED/OK
>--System Board Fan4: 3480 RPM - ENABLED/OK
>--System Board Fan5: 3600 RPM - ENABLED/OK
>--System Board Fan6: 3600 RPM - ENABLED/OK
>BATTERY
>--System Board CMOS Battery: ENABLED/OK [PRESENCEDETECTED]
>--PERC1 ROMB Battery: ENABLED/OK [PRESENCEDETECTED]
>PU
>--PU 1: ENABLED/OK, RedundancyStatus: FULL, SystemBoard Pwr Consumption: 182 W
>MEM
>--Memory 1 (DIMM Socket A1) 8.0 GB/2133 MHz: ENABLED/OK [26, Samsung, S/N: ********]
>--Memory 2 (DIMM Socket A2) 8.0 GB/2133 MHz: ENABLED/OK [26, Samsung, S/N: ********]
>--Memory 3 (DIMM Socket A3) 8.0 GB/2133 MHz: ENABLED/OK [26, Samsung, S/N: ********]
>--Memory 4 (DIMM Socket A4) 8.0 GB/2133 MHz: ENABLED/OK [26, Samsung, S/N: ********]
>--Memory 5 (DIMM Socket B1) 8.0 GB/2133 MHz: ENABLED/OK [26, Samsung, S/N: ********]
>--Memory 6 (DIMM Socket B2) 8.0 GB/2133 MHz: ENABLED/OK [26, Samsung, S/N: ********]
>--Memory 7 (DIMM Socket B3) 8.0 GB/2133 MHz: ENABLED/OK [26, Samsung, S/N: ********]
>--Memory 8 (DIMM Socket B4) 8.0 GB/2133 MHz: ENABLED/OK [26, Samsung, S/N: ********]
>VDISK
>--VDisk 1 (System): OK/ONLINE, RAID-1 (185.75 GB), BadBlock: 0 [Virtual Disk 0 on Integrated RAID Controller 1]
>--VDisk 2 (Backup2Disk): OK/ONLINE, RAID-6 (7820.75 GB), BadBlock: 0 [Virtual Disk 1 on Integrated RAID Controller 1]
>--VDisk 3 (Datadepot): OK/ONLINE, RAID-5 (4469.0 GB), BadBlock: 0 [Virtual Disk 2 on Integrated RAID Controller 1]
>DISK
>--PDisk 1 (0:1:0) 1117.25 GB: ONLINE, PowerStat: SPUNUP, HotSpare: no [SEAGATE, S/N: ********]
>--PDisk 2 (0:1:1) 1117.25 GB: ONLINE, PowerStat: SPUNUP, HotSpare: no [SEAGATE, S/N: ********]
>--PDisk 3 (0:1:2) 1117.25 GB: ONLINE, PowerStat: SPUNUP, HotSpare: no [SEAGATE, S/N: ********]
>--PDisk 4 (0:1:3) 1117.25 GB: ONLINE, PowerStat: SPUNUP, HotSpare: no [SEAGATE, S/N: ********]
>--PDisk 5 (0:1:4) 1117.25 GB: ONLINE, PowerStat: SPUNUP, HotSpare: no [SEAGATE, S/N: ********]
>--PDisk 6 (0:1:5) 1117.25 GB: ONLINE, PowerStat: SPUNUP, HotSpare: no [SEAGATE, S/N: ********]
>--PDisk 7 (0:1:6) 1117.25 GB: ONLINE, PowerStat: SPUNUP, HotSpare: no [SEAGATE, S/N: ********]
>--PDisk 8 (0:1:7) 1117.25 GB: ONLINE, PowerStat: SPUNUP, HotSpare: no [SEAGATE, S/N: ********]
>--PDisk 9 (0:1:8) 1117.25 GB: ONLINE, PowerStat: SPUNUP, HotSpare: no [SEAGATE, S/N: ********]
>--PDisk 10 (0:1:12) 1117.25 GB: ONLINE, PowerStat: SPUNUP, HotSpare: no [SEAGATE, S/N: ********]
>--PDisk 11 (0:1:13) 1117.25 GB: ONLINE, PowerStat: SPUNUP, HotSpare: no [SEAGATE, S/N: ********]
>--PDisk 12 (0:1:14) 1117.25 GB: ONLINE, PowerStat: SPUNUP, HotSpare: no [SEAGATE, S/N: ********]
>--PDisk 13 (0:1:15) 1117.25 GB: ONLINE, PowerStat: SPUNUP, HotSpare: no [SEAGATE, S/N: ********]
>--PDisk 14 (0:1:16) 1117.25 GB: ONLINE, PowerStat: SPUNUP, HotSpare: no [SEAGATE, S/N: ********]
>--PDisk 15 (0:1:23) 1117.25 GB: READY, PowerStat: SPUNUP, HotSpare: no [SEAGATE, S/N: ********]
>--PDisk 16 (0:1:24) 185.75 GB: ONLINE, PowerStat: 5(!), HotSpare: no [TOSHIBA, S/N: ************]
>--PDisk 17 (0:1:25) 185.75 GB: ONLINE, PowerStat: 5(!), HotSpare: no [TOSHIBA, S/N: ************]
>SENSOR
>--System Board Inlet Temp: 27.0 C ENABLED/OK
>--System Board Exhaust Temp: 38.0 C ENABLED/OK
>--CPU1 Temp: 44.0 C ENABLED/OK
>--CPU2 Temp: 47.0 C ENABLED/OK
>CPU
>--CPU 1 (8 cores/16 threads): ENABLED/OK [Intel(R) Xeon(R) CPU E5-2630L v3 @ 1.80GHz]
>--CPU 2 (8 cores/16 threads): ENABLED/OK [Intel(R) Xeon(R) CPU E5-2630L v3 @ 1.80GHz]

Also a of my iDRAC Web GUI at the PDisk 16 (0:1:24):
![idrac](https://cloud.githubusercontent.com/assets/6261420/12292283/4c8b1724-b9ec-11e5-880b-7a86fd232ef9.jpg)

For any help solving the problem thanks in advance!

Greetings from Germany 
Nick

**Comment by dangmocrang** [comment]
Hi, i only check the return snmp string, in this case your idrac return with value "5" which differ from other disks. So you need to check your idrac help page and see if "5" status is ok or not. If its ok then in idrac.conf, at section "OK", add "5".

**Comment by dangmocrang** [comment]
'''
OK = ok|online|spunup|full|ready|enabled|presence|5
'''

**Comment by quicknick** [comment]
Perfect, that did the trick, so many thanks for the workaround :-) 

Sadly, there is no Status explanation for status 5 in idrac help.

**Comment by adorsey-NOAA** [comment]
I apologize for commenting on an old issue, but I wanted to provide an update.

I ran into this same issue today with a Dell PowerEdge R740xd with several SSDs installed.  The SSDs had the unknown status 5 as shown above.

I went to https://www.dell.com/support/home/us/en/04/drivers/driversdetails?driverId=PNHK2 and downloaded the latest MIBS, then copied the new iDRAC MIBs to /usr/share/snmp/mibs/.

This fixed the issue - the SSDs are no longer reporting as failed.  (Status 5 is 'on', it was likely added for SSDs that aren't technically 'spun up'.)
